wagtail/tests/test_tests.py

```diff
@@ -172,6 +172,15 @@ def test_assert_can_create(self):
             },
         )
 
+    def test_assert_can_create_for_unpublished_page(self):
+        with self.assertRaises(self.failureException):
+            self.assertCanCreate(
+                self.root,
+                SimplePage,
+                {"title": "Simple Lorem Page", "content": "Lorem ipsum dolor sit amet"},
+                publish=False,
+            )
+
```

Comment:
Sorry for the confusion here, I think there is a bit of a mix up - we do want to test the assertion works but when `publish is True` (just to ensure the test works as expected) I will fix this up as follows.

```
    def test_assert_can_create_for_page_with_publish(self):
        self.assertCanCreate(
            self.root,
            SimplePage,
            {"title": "Simple Lorem Page", "content": "Lorem ipsum dolor sit amet"},
            publish=True,
        )
```

See the `publish=True` - this is the code path we want to check works.